CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+### Contributor License Agreements
+
+We'd love to accept your patches! Before we can take them, we have to jump a
+couple of legal hurdles.
+
+The Tyk CLA [must be signed](https://github.com/TykTechnologies/tyk/blob/master/CLA.md) by all contributors. You will be automatically asked to sign CLA once PR will be created.
+
+Once you are CLA'ed, we'll be able to accept your pull requests. For any issues that you face during this process, please create an GitHub issue explaining the problem and we will help get it sorted out.
+
+***NOTE***: Only original source code from you and other people that have
+signed the CLA can be accepted into the repository. This policy does not
+apply to [vendor](vendor/).
+
+### Finding Things That Need Help
+
+If you're new to the project and want to help, but don't know where to start,
+we have a semi-curated list of issues that have should not need deep knowledge
+of the system.  [Have a look and see if anything sounds
+interesting](https://github.com/TykTechnologies/tyk/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted).
+
+Alternatively, read some of the [many docs on the system](https://tyk.io/tyk-documentation/get-started/), and pick a component that seems
+interesting.  Start with `main()` and read
+until you find something you want to fix.  The best way to learn is to hack!
+There's always code that can be clarified and variables or functions that can
+be renamed or commented.
+
+### `master` is unsafe
+
+We will do our best to keep master in good shape, with tests passing at all times. But in order to move fast, we will make API changes that your application might not be compatible with. We will do our best to communicate these changes and version appropriately so you can lock into a specific version if need be. For stable releases check our tags and `stable` branch, which contains our latest stable release.
+
+
+### Contributing A Patch
+
+If you're working on an existing issue, such as one of the `help-wanted` ones
+above, simply respond to the issue and express interest in working on it.  This
+helps other people know that the issue is active, and hopefully prevents
+duplicated efforts.
+
+If you want to work on a new idea of relatively small scope:
+
+1. Submit an issue describing your proposed change to the repo in question.
+1. The repo owners will respond to your issue promptly.
+1. Clone the repo, develop, and test your changes.
+1. Submit a pull request.
+1. If your proposed change is accepted, and you haven't already done so, sign a
+   Contributor License Agreement (see details above).
+
+If you want to work on a bigger idea, we STRONGLY recommend that you start with
+some bugs or smaller features. It is always better to discuss your idea with our team first, before implementing it.
+
+### Downloading the project
+You need to clone Tyk from GitHub to your GOPATH folder, or alternatively you can run `go get -d github.com/TykTechnologies/tyk` which automatically downloads project to the right path.
```

Comment:
I'm not sure why, but right now with "-u" it generate lot of warnings like this:

```
package github.com/TykTechnologies/tyk/vendor/gopkg.in/vmihailenco/msgpack.v2: github.com/TykTechnologies/tyk is a custom import path for https://github.com/TykTechnologies/tyk, but /Users/buger/Documents/work/go/src/github.com/TykTechnologies/tyk is checked out from https://github.com/TykTechnologies/tyk.git
```